osg/short-term.md

```diff
@@ -0,0 +1,292 @@
+    * [Sending Job Status Updates](#sending-job-status-updates)
+        * [Status Codes](#status-codes)
+    * [Retrieving Input Files](#retrieving-input-files)
+    * [Running the Job](#running-the-job)
+    * [Uploading Output Files](#uploading-output-files)
+    * [Caveats](#caveats)
+
+
+# Short-Term OSG Integration
+
+The DE can submit jobs to OSG, but the current integration mechanism is very limited in scope and places much of the
+burden on the app integrator. For OSG jobs, the DE performs the following tasks:
+
+- obtains iRODS read tickets for each of the job input files and creates a file listing the tickets and paths
+- obtains an iRODS write ticket for the job output folder and creates a file listing the tickets and paths
+- creates a JSON configuration file containing information needed to run the job
+- submits the job to OSG
+
+The app integrator is expected to provide a wrapper script that performs the following tasks:
+
+- reads and parses the JSON configuration file
+- initializes the iRODS connection based on information available in the configuration file
+- sends job status updates back to the DE using a URL provided in the configuration file
+- parses the input ticket list file and retrieves all of the input files from iRODS
+- runs the job using the arguments provided in the configuration file
+- redirects job output to the file names provided in the configuration file
+- parses the output ticket list file and uploads all job outputs to iRODS
+
+## Image Overview
+
+OSG creates the Singularity images in its CVM-FS repository by pulling Docker images from Dockerhub and converting them
+to Singularity images. This means that the first step to create a Singularity image for OSG is to create a Docker image
+and push it to Dockerhub. Instructions for getting Docker images into OSG's CVM-FS repository can be found [here](1).
+
+### Requirements
+
+The primary requirements are that the image must contain an executable file called `/usr/bin/wrapper`, and this program
+must read its configuration settings from a file called `config.json` in the current working directory. The
+implemenation language is unimportant as long as all of the tasks required of the wrapper can be performed. CyVerse has
+a reference implementation of such a wrapper script implemented in Python 2.7 available [here](2).
+
+Each image must also have a way to communicate with iRODS. Usually, this means going with one of two options. If you're
+using a JVM language such as Java, Scala or Clojure for the wrapper script then you can communicate with iRODS using
+Jargon. If you're not using a JVM language then the easiest option is to use the iRODS icommands. The wrapper script
+reference implementation provided by CyVerse uses the icommands which are installed as a step in the [corresponding
+Dockerfile](3).
+
+This image should also contain an empty directory called `/cvmfs`. When the job is executed, OSG's CVM-FS repository
+will be mounted to this path.
```

Comment:
Does the Dockerfile (and corresponding image) also need to have a working directory pre-created? I'm not quite sure what gets mounted where by condor's singularity universe, but if it does need one we may as well mention it here too.

Reply:
I don't believe that it has to have a specific working directory. At least the CentOS images used by OSG use the default working directory in the base CentOS images. I'll probably add a note to this effect, though.